client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -227,6 +232,31 @@ class DeltaSharingRestClient(
       .setConnectTimeout(timeoutInSeconds * 1000)
       .setConnectionRequestTimeout(timeoutInSeconds * 1000)
       .setSocketTimeout(timeoutInSeconds * 1000).build()
+    proxyConfigOpt.foreach { proxyConfig =>
+      if (sslTrustAll) {
+        throw new IllegalStateException(
+          "Proxy configuration is not supported when sslTrustAll is enabled.")
+      }
+      val proxy = new HttpHost(proxyConfig.host, proxyConfig.port)
+      clientBuilder.setProxy(proxy)
+
+      if (proxyConfig.noProxyHosts.nonEmpty) {
+        val routePlanner = new DefaultRoutePlanner(DefaultSchemePortResolver.INSTANCE) {
+          override def determineRoute(target: HttpHost,
+                                      request: HttpRequest,
+                                      context: HttpContext): HttpRoute = {
+            if (proxyConfig.noProxyHosts.contains(target.getHostName)) {
+              // Direct route (no proxy)
+              new HttpRoute(target)
+            } else {
+              // Route via proxy
+              new HttpRoute(target, proxy)
```

Comment:
This constructor always constructs an insecure (http) route through the proxy. This does not work if one is using an **http** proxy to make an **https** call to the server. Instead, consider using an alternate constructor and setting the `secure` flag appropriately based on the target:

```suggestion
              val isSecure = target.getSchemeName == "https"
              new HttpRoute(target, null, proxy, isSecure)
```